core/src/test/java/org/apache/druid/math/expr/ParserTest.java

```diff
@@ -179,24 +183,262 @@ public void testLiterals()
     validateConstantExpression("\'f\\u000Ao \\'b\\\\\\\"ar\'", "f\no 'b\\\"ar");
   }
 
+  @Test
+  public void testLiteralArrays()
+  {
+    validateConstantExpression("[1.0, 2.345]", new Double[]{1.0, 2.345});
+    validateConstantExpression("[1, 3]", new Long[]{1L, 3L});
+    validateConstantExpression("[\'hello\', \'world\']", new String[]{"hello", "world"});
+  }
+
   @Test
   public void testFunctions()
   {
     validateParser("sqrt(x)", "(sqrt [x])", ImmutableList.of("x"));
     validateParser("if(cond,then,else)", "(if [cond, then, else])", ImmutableList.of("cond", "then", "else"));
+    validateParser("cast(x, 'STRING')", "(cast [x, STRING])", ImmutableList.of("x"));
+    validateParser("cast(x, 'LONG')", "(cast [x, LONG])", ImmutableList.of("x"));
+    validateParser("cast(x, 'DOUBLE')", "(cast [x, DOUBLE])", ImmutableList.of("x"));
+    validateParser(
+        "cast(x, 'STRING_ARRAY')",
+        "(cast [x, STRING_ARRAY])",
+        ImmutableList.of("x"),
+        ImmutableSet.of(),
+        ImmutableSet.of("x")
+    );
+    validateParser(
+        "cast(x, 'LONG_ARRAY')",
+        "(cast [x, LONG_ARRAY])",
+        ImmutableList.of("x"),
+        ImmutableSet.of(),
+        ImmutableSet.of("x")
+    );
+    validateParser(
+        "cast(x, 'DOUBLE_ARRAY')",
+        "(cast [x, DOUBLE_ARRAY])",
+        ImmutableList.of("x"),
+        ImmutableSet.of(),
+        ImmutableSet.of("x")
+    );
+    validateParser(
+        "array_length(x)",
+        "(array_length [x])",
+        ImmutableList.of("x"),
+        ImmutableSet.of(),
+        ImmutableSet.of("x")
+    );
+    validateParser(
+        "array_concat(x, y)",
+        "(array_concat [x, y])",
+        ImmutableList.of("x", "y"),
+        ImmutableSet.of(),
+        ImmutableSet.of("x", "y")
+    );
+    validateParser(
+        "array_append(x, y)",
+        "(array_append [x, y])",
+        ImmutableList.of("x", "y"),
+        ImmutableSet.of("y"),
+        ImmutableSet.of("x")
+    );
+
+    validateFlatten("sqrt(4)", "(sqrt [4])", "2.0");
+    validateFlatten("array_concat([1, 2], [3, 4])", "(array_concat [[1, 2], [3, 4]])", "[1, 2, 3, 4]");
   }
 
+  @Test
+  public void testApplyFunctions()
+  {
```

Comment:
add a case for no arg ..

```suggestion
  {
    validateParser(
        "map(() -> 1, x)",
        "(map ([] -> 1), [x])",
        ImmutableList.of("x"),
        ImmutableSet.of(),
        ImmutableSet.of("x")
    );
```